mesh/meshdb.go

```diff
@@ -1076,3 +1165,29 @@ func (db *BlockFetcherDB) Get(hash []byte) ([]byte, error) {
 
 	return data, nil
 }
+
+// newBallotFetcherDB returns reference to a BallotFetcherDB instance.
+func newBallotFetcherDB(mdb *DB) *BallotFetcherDB {
+	return &BallotFetcherDB{mdb: mdb}
+}
+
+// BallotFetcherDB implements API that allows fetcher to get a ballot from a remote database.
+type BallotFetcherDB struct {
+	mdb *DB
+}
+
+// Get types.Block encoded in byte using hash.
+func (db *BallotFetcherDB) Get(hash []byte) ([]byte, error) {
+	id := types.BallotID(types.BytesToHash(hash).ToHash20())
+	blk, err := db.mdb.GetBallot(id)
+	if err != nil {
+		return nil, fmt.Errorf("get ballot: %w", err)
+	}
+
+	data, err := types.InterfaceToBytes(blk)
```

Comment:
can we use codec.Encode / codec.Decode in the new code please

Reply:
done for the files in this PR